includes/admin/beta-features.php

```diff
@@ -0,0 +1,316 @@
+		 * Contains an array of admin beta feature instances.
+		 *
+		 * @var array
+		 */
+		private $beta_features = array();
+
+		/**
+		 * The active beta feature.
+		 *
+		 * @var string
+		 */
+		private $active = '';
+
+		/**
+		 * This function will setup the class functionality
+		 *
+		 * @since   SCF 6.4.2
+		 *
+		 * @return  void
+		 */
+		public function __construct() {
+			// Temporarily disabled - will be enabled when beta feature is ready
+			// add_action( 'admin_menu', array( $this, 'admin_menu' ), 20 );
+		}
+
+		/**
+		 * This function will store an beta feature class instance in the beta features array.
+		 *
+		 * @since   SCF 6.4.2
+		 *
+		 * @param   string $beta_feature Class name.
+		 * @return  void
+		 */
+		public function register_beta_feature( $beta_feature ) {
+			$instance                               = new $beta_feature();
+			$this->beta_features[ $instance->name ] = $instance;
+		}
+
+		/**
+		 * This function will return an beta feature class or null if not found.
+		 *
+		 * @since   SCF 6.4.2
+		 *
+		 * @param   string $name Name of beta feature.
+		 * @return  mixed (SCF_Admin_Beta_Feature|null)
+		 */
+		public function get_beta_feature( $name ) {
+			return isset( $this->beta_features[ $name ] ) ? $this->beta_features[ $name ] : null;
+		}
+
+		/**
+		 * This function will return an array of all beta feature instances.
+		 *
+		 * @since   SCF 6.4.2
+		 *
+		 * @return  array
+		 */
+		public function get_beta_features() {
+			// Include beta features
+			$this->include_beta_features();
+
+			return $this->beta_features;
+		}
+
+		/**
+		 * Localizes the beta features data.
+		 *
+		 * @since   SCF 6.4.2
+		 *
+		 * @return  void
+		 */
+		public function localize_beta_features() {
+			$beta_features = array();
+			foreach ( $this->get_beta_features() as $name => $beta_feature ) {
+				$beta_features[ $name ] = $beta_feature->is_enabled();
+			}
+
+			acf_localize_data(
+				array(
+					'betaFeatures' => $beta_features,
+				)
+			);
+		}
+
+		/**
+		 * This function will add the SCF beta features menu item to the WP admin
+		 *
+		 * @type    action (admin_menu)
+		 * @since   SCF 6.4.2
+		 *
+		 * @return  void
+		 */
+		public function admin_menu() {
+			// bail early if no show_admin
+			if ( ! acf_get_setting( 'show_admin' ) ) {
+				return;
+			}
+
+			$page = add_submenu_page( 'edit.php?post_type=acf-field-group', __( 'Beta Features', 'secure-custom-fields' ), __( 'Beta Features', 'secure-custom-fields' ), acf_get_setting( 'capability' ), 'scf-beta-features', array( $this, 'html' ) );
+
+			add_action( 'load-' . $page, array( $this, 'load' ) );
+			add_action( 'admin_enqueue_scripts', array( $this, 'localize_beta_features' ), 20 );
+		}
+
+		/**
+		 * Loads the admin beta features page.
+		 *
+		 * @since   SCF 6.4.2
+		 *
+		 * @return  void
+		 */
+		public function load() {
+			add_filter( 'admin_body_class', array( $this, 'admin_body_class' ) );
+			// disable filters (default to raw data)
```

Comment:
What's the reason to disable the filters?

Reply:
It was initally added to prevent in a future with data that can be filtered by different plugins. It is more a security method, as we won't know how this new features will work with different scf extensions, but for now we can remove it too.